python/ray/serve/batching.py

```diff
@@ -183,34 +191,51 @@ async def _consume_func_generator(
                     elif result in [StopIteration, StopAsyncIteration]:
```

Comment:
StopIteration will be thrown, not returned, right? This part is confusing

Reply:
No, `StopIteration` must be returned, not thrown. This is part of the API for batched streaming in Serve.

For context, static batching allows the generators in the batch to have different lengths. To terminate requests in the batch that are finished, the batch handler can return a `StopIteration` object as the output for those requests. See [the docs](https://docs.ray.io/en/releases-2.9.0/serve/advanced-guides/dyn-req-batch.html#streaming-batched-requests) for more info.

Reply:
Oh, got it. Can you please add a comment elaborating that this is Serve's convention? It looks quite confusing otherwise

Reply:
yeah can see how this is confusing... it was the best of multiple bad options when we decided on it :)

Reply:
I made the list a global variable and added a comment alongside it. Let me know if that clarifies it.